template/cordova/lib/package.js

```diff
@@ -130,58 +145,47 @@ module.exports.findDevice = function (target) {
-            }));
+module.exports.listDevices = function (deploymentTool) {
+    
+    return deploymentTool.enumerateDevices().then(function(deviceList) {
+        return deviceList.map(function(device) {
+            return device.toString();
         });
-    });
-};
-
 
-function installAppToPhone(appDeployUtils, package, target, update) {
-    // /installlaunch option sometimes fails with 'Error: The parameter is incorrect.'
-    // so we use separate steps to /install or /update and then /launch
-    var cmd = update ? '/update' : '/install';
-    console.log('Installing application...');
-    return spawn(appDeployUtils, [cmd, package.appx, '/targetdevice:' + target]);
-}
+    }, function(e) {
+        console.warn('Failed to enumerate devices');
+        console.warn(e);
 
-function runAppOnPhone(appDeployUtils, target) {
-    return Q().then(function() {
-        return module.exports.getAppId(path.join(__dirname, '..', '..'));
-    }).then(function(appId) {
-        console.log('Running application... ');
-        return spawn(appDeployUtils, ['/launch', appId, '/targetdevice:' + target]);
+        throw e;
     });
-}
+};
+
 
 function uninstallAppFromPhone(appDeployUtils, package, target) {
     console.log('Attempting to remove previously installed application...');
-    return spawn(appDeployUtils, ['/uninstall', package.phoneId, '/targetdevice:' + target]);
+    return appDeployUtils.uninstallAppPackage(package.phoneId, target);
 }
 
 // deploys specified phone package to device/emulator and launches it
-module.exports.deployToPhone = function (package, deployTarget, targetWindows10) {
-    var getTarget = deployTarget == 'device' ? Q('de') :
-        deployTarget == 'emulator' ? Q('xd') : module.exports.findDevice(deployTarget);
-
-    return getTarget.then(function(target) {
-        return utils.getAppDeployUtils(targetWindows10).then(function(appDeployUtils) {
+module.exports.deployToPhone = function (package, deployTarget, targetWindows10, deploymentTool) {
+    var deployment;
+    if (deploymentTool) {
+        deployment = Q(deploymentTool);
+    }
+    else {
+        deployment = utils.getAppDeployUtils(targetWindows10);
+    }
 
-            return uninstallAppFromPhone(appDeployUtils, package, target).then(
+    return deployment.then(function(deploymentTool) {
+        return module.exports.findDevice(deploymentTool, deployTarget).then(function(target) {
+            return uninstallAppFromPhone(deploymentTool, package, target).then(
                 function() {}, function() {}).then(function() {
-                    return installAppToPhone(appDeployUtils, package, target, false);
-                }).then(function() {
-                    return runAppOnPhone(appDeployUtils, target);
-                }, function(error) {
+                    // shouldUpdate = false because we've already uninstalled
+                    return deploymentTool.installAppPackage(package.appx, target, true, false);
```

Comment:
I don't understand this comment.  Code comments should specify why you do something when it isn't immediately clear, not just repeat the code in comment form.